File: src/nl/rubensten/texifyidea/structure/filter/BibitemFilter.java
package nl.rubensten.texifyidea.structure.filter;

import com.intellij.ide.util.treeView.smartTree.ActionPresentation;
import com.intellij.ide.util.treeView.smartTree.Filter;
import com.intellij.ide.util.treeView.smartTree.TreeElement;
import nl.rubensten.texifyidea.TexifyIcons;
import nl.rubensten.texifyidea.structure.latex.LatexStructureViewCommandElement;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;

/**
 * @author Ruben Schellekens
 */
public class BibitemFilter implements Filter {

    @Override
    public boolean isVisible(TreeElement treeElement) {
        if (!(treeElement instanceof LatexStructureViewCommandElement)) {
            return true;
        }

        LatexStructureViewCommandElement element = (LatexStructureViewCommandElement)treeElement;
        return !element.getCommandName().equals("\\bibitem");
    }

    @Override
    public boolean isReverted() {
        return true;
    }

    @NotNull
    @Override
    public ActionPresentation getPresentation() {
        return BibitemFilterPresentation.INSTANCE;
    }

    @NotNull
    @Override
    public String getName() {
        return "latex.texify.filter.bibitem";
    }

    /**
     * @author Ruben Schellekens
     */
    private static class BibitemFilterPresentation implements ActionPresentation {

        private static final BibitemFilterPresentation INSTANCE = new BibitemFilterPresentation();

        @NotNull
        @Override
        public String getText() {
            return "Show Bibliography Items";
        }

        @Override
        public String getDescription() {
            return "Show Bibliography Items";
        }

        @Override
        public Icon getIcon() {
            return TexifyIcons.DOT_BIB;
        }
    }
}

File: src/nl/rubensten/texifyidea/structure/filter/BibitemFilter.kt
package nl.rubensten.texifyidea.structure.filter

import com.intellij.ide.util.treeView.smartTree.ActionPresentation
import com.intellij.ide.util.treeView.smartTree.Filter
import com.intellij.ide.util.treeView.smartTree.TreeElement
import nl.rubensten.texifyidea.TexifyIcons
import nl.rubensten.texifyidea.structure.latex.LatexStructureViewCommandElement

/**
 * @author Ruben Schellekens
 */
class BibitemFilter : Filter {

    override fun isVisible(treeElement: TreeElement): Boolean {
        return if (treeElement !is LatexStructureViewCommandElement) {
            true
        }
        else treeElement.commandName != "\\bibitem"
    }

    override fun isReverted() = true

    override fun getPresentation() = BibitemFilterPresentation

    override fun getName() = "latex.texify.filter.bibitem"

    /**
     * @author Ruben Schellekens
     */
    object BibitemFilterPresentation : ActionPresentation {

        override fun getText() = "Show Bibliography Items"

        override fun getDescription() = "Show Bibliography Items"

        override fun getIcon() = TexifyIcons.DOT_BIB!!
    }
}